cpp/src/arrow/status.cc

```diff
@@ -134,9 +134,8 @@ void Status::Abort(const std::string& message) const {
 void Status::Warn() const { Warn(std::string()); }
 
 void Status::Warn(const std::string& message) const {
-  ARROW_LOG(WARNING) << "-- Arrow Warning --"
-                     << message    << "\n"
-                     << ToString() << "\n";
+  ARROW_LOG(WARNING) << "-- Arrow Warning --\n"
+                     << message << "\n" << ToString();
```

Comment:
maybe I should remove the "\n" and replace with ": "?

I also kept the header (`-- Arrow Warning --`) in the same style as the abort message, but maybe it'd be nice to change it to `[Arrow Warning] -- ` and then remove that newline as well?